frame/bounties/src/subbounty_migration.rs

```diff
@@ -0,0 +1,166 @@
+use sp_std::prelude::*;
+use frame_support::{
+	weights::Weight, Twox64Concat,
+	storage::types::{StorageMap},
+	crate_to_pallet_version,
+};
+
+use sp_runtime::RuntimeDebug;
+
+/// An index of a bounty. Just a `u32`.
+pub type BountyIndex = u32;
+
+/// Trait to implement to give information about types used for migration
+pub trait SubBountyMigration {
+
+	type AccountId: 'static + FullCodec;
+
+	type BlockNumber: 'static + FullCodec;
+
+	type Balance: 'static + FullCodec + Copy;
+}
+
+/// The status of a bounty proposal.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum BountyStatus<AccountId, BlockNumber> {
+	/// The bounty is proposed and waiting for approval.
+	Proposed,
+	/// The bounty is approved and waiting to become active at next spend period.
+	Approved,
+	/// The bounty is funded and waiting for curator assignment.
+	Funded,
+	/// A curator has been proposed by the `ApproveOrigin`. Waiting for acceptance from the curator.
+	CuratorProposed {
+		/// The assigned curator of this bounty.
+		curator: AccountId,
+	},
+	/// The bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// An update from the curator is due by this block, else they are considered inactive.
+		update_due: BlockNumber,
+	},
+	/// The bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// The beneficiary of the bounty.
+		beneficiary: AccountId,
+		/// When the bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct OldBounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct Bounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+	/// active Subbounty count
+	active_subbounty_count: BountyIndex,
+}
+
+struct StorageMigrationBounties;
+impl frame_support::traits::StorageInstance for StorageMigrationBounties {
+	fn pallet_prefix() -> &'static str { "Treasury" }
+	const STORAGE_PREFIX: &'static str = "Bounties";
+}
```

Comment:
You can use `generate_storage_alias` for this.

Reply:
I agree the different pallet is having different Storage migration code, Storage Migration for this PR, I took reference from @thiolliere, existing implementation is discussed & reviewed by @thiolliere. If its too much deviated, Please let me know will take it as separate PR.

Reply:
I'm ok to use generate_storage_alias, but StorageInstance should be quite stable so the boilerplate shouldn't be maintainance burden.